C/common/readingset_circularbuffer.cpp

```diff
@@ -98,8 +98,8 @@ void ReadingSetCircularBuffer::appendReadingSet(const std::vector<Reading *>& re
 std::vector<std::shared_ptr<ReadingSet>> ReadingSetCircularBuffer::extract(bool isExtractSingleElement)
 {
 	
-    std::lock_guard<std::recursive_mutex> guard(m_mutex);
-    bool isUnreadDataInBuffer = isEmpty() || (m_nextReadIndex == m_circularBuffer.size());
+    lock_guard<mutex> guard(m_mutex);
+    bool isUnreadDataInBuffer = m_circularBuffer.empty() || (m_nextReadIndex == m_circularBuffer.size());
```

Comment:
Your variable name has the opposite meaning. You are checking whether m_circularBuffer doesn't have unread data. May be invert the boolean condition on the right hand side or rename the variable.